### DISCUSSION TITLE
Feature request: Implement option "Click on empty space to deselect items"

### DISCUSSION THREAD

**Discussion by WhiteChairFromIkea** [doublecmd/doublecmd#1162] [Suggestions]
It is usual in other file explorers to click on empty space in order to deselect files.

In DC, I have added button with action "cm_MarkUnmarAll" in order to easily/cuickly deselect files, this seems to be overkill :)

O suggest to implement additional checkbox in preferences "Click on empty space to deselect items"

```
Double Commander
Version: 1.0.0 alpha
Revision: 9890
Commit: 5b000ea17
Build date: 2021/08/22
Lazarus: 2.2.0.2
FPC: 3.0.4
Platform: x86_64-Win64-qt5
OS version: Windows 10 Home 20H2 x86_64

Widgetset library: Qt 5.13.0, libQt5Pas 5.6.1
```

**Comment by Skif-off** [comment]
> It is usual in other file explorers

Did you mean "other dual-pane managers"?

> In DC, I have added button with action "cm_MarkUnmarAll" in order to easily/cuickly deselect files

Item ".." works fine. For example, other ways:
- Configuration > Options... > Mouse > Selection by mouse > enable "By clicking on icon": the first click will unmark one file, now click on his name.
- Use Ctrl+mouse click: it will unmark one file, now click on his name.

**Comment by WhiteChairFromIkea** [comment]
>>It is usual in other file explorers

>Did you mean "other dual-pane managers"?

I don't understand this question. I meant to say MS default explorer.exe, Windows "Double commander", Linux "Nemo" also does this, two last explorers are dual pane, used them before switching to DC. DC is a black sheep here, probably  because of historical reasons and it is pretty hard to accommodate for nowadays user (see https://github.com/doublecmd/doublecmd/issues/167). Also see https://doublecmd.h1n.ru/viewtopic.php?t=4282, I am not alone.

> Item ".." works fine. For example, other ways:

This is a good workaround, thank you. But not easily discoverable, didn't found that for using DC for several months.

----

All in all, I am going to implement this option anyway.

**Comment by WhiteChairFromIkea** [comment]
@alexx2000 , at which preferences page checkbox "Clicking on empty space removes selection" should be added to?

**Comment by alexx2000** [comment]
It can be added to "Options"->"File views"->"File views extra". But it's not very useful because DC has empty space only when file panel has small file count.

**Reply by vertigo220** [reply]
I don't understand this. I have tons of empty space even in directories with hundreds of folders/files, and all you need is a small gap at the right edge, which I always have for uses like this and for selecting multiple files with the mouse.

This issue is related to a general issue I've found with DC regarding file selection. Perhaps it's because DC isn't a "traditional" (to Windows users) file manager. Maybe DC just isn't going to work for those of us that are used to file managers working a certain way, which would be a shame because, as I've mentioned before, it's one of a very few managers that's cross-platform, and one of the better and more functional ones, and therefore could be a great bridge for people looking to leave Windows (e.g. I have no intention of staying with Windows past Win10's EOL in just under a year, so I'm looking at this as something I can use in the meantime and continue to use once transitioning, especially since it seems to be one of the best for Linux as well). It seems there's a lot of resistance to making it work contrary to the orthodox design and behave more like a navigational file manager. Perhaps making it do both is too difficult, though it does seem possible as DOpus and XTree are listed as both, but if it can be done, it would open up a great option to a LOT of users that are more familiar with navigational managers, mainly Windows File Explorer and, to a lesser extent, Dolphin, DOpus, Nautilus, and macOS Finder.

Anyways, I find DC very confusing, as I had a file selected (I left-clicked it with the mouse and it was highlighted), yet the status bar showed 0 selected. I clicked in empty space and nothing changed. I ctrl-clicked on another file and suddenly there were 2 selected (which is what I would expect, except now this almost seems backwards because the first file wasn't selected, but now is when I click _another_ file), and again clicking in empty space did nothing, nor did clicking on one of the selected files, and only clicking on a _different_ file worked to deselect those files. Again, I assume this is "normal" behavior for orthodox managers, but for those used to more common navigational ones, this is very unusual. I'd love it if DC could optionally act in either way, to allow both groups to use it.

**Comment by WhiteChairFromIkea** [comment]
> But it's not very useful because DC has empty space only when file panel has small file count.

This is an exact usual working scenario for me (small file count)  :)

**Comment by Skif-off** [comment]
@WhiteChairFromIkea 

> I don't understand this question. I meant to say MS default explorer.exe, Windows "Double commander", Linux "Nemo" also does this, two last explorers are dual pane, used them before switching to DC.

The answer is [here](https://github.com/doublecmd/doublecmd/issues/167#issuecomment-906988815), directly сompare Windows Explorer, Nemo, Thunar and so on with Double Commander, Total Commander, Krusader and other is simply incorrect, they are different.

> This is a good workaround, thank you. But not easily discoverable, didn't found that for using DC for several months.

I don't know, I had no problem with it. I willl think about how to add it in FAQ.

**Comment by Skif-off** [comment]
@alexx2000
Are you planning to add this feature? If true, could you turn it off by default? Typical use of single click is activation of file panel or DC main window...

**Comment by alexx2000** [comment]
May be. Yes it should be off by default. If panel is inactive then first click should activate it and only second clear selection.
